### PR TITLE
Feature/basic drive motor control

### PR DESCRIPTION
**Pull request by emcauliffe**

### CODE REVIEW

**Comment by emcauliffe** [comment]
Need to handle issues with integer overflow

**Comment by emcauliffe** [comment]
Need to increase max rover speed from 4 to 50 (0.1*500), or provide 500 steps of speed granularity from drive control systems. STM32 timers allow for 500 steps of granularity so we should use them to their full potential.

**Comment by emcauliffe** [comment]
> Need to increase max rover speed from 4 to 50 (0.1*500), or provide 500 steps of speed granularity from drive control systems. STM32 timers allow for 500 steps of granularity so we should use them to their full potential.


> Need to handle issues with integer overflow

implemented fixes for these

**Comment by emcauliffe** [comment]
> Need to handle issues with integer overflow

File: firmware/dmc_stm32/drive_test.ino [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ros::NodeHandle_<NewHardware> nh;
+
+// subscribe to cmd_vel Twist topic
+void controlMotors(const geometry_msgs::Twist &cmd_vel);
+ros::Subscriber<geometry_msgs::Twist> velocity("cmd_vel", &controlMotors);
+
+void setup()
+{
+    configureHardwareTimers();
+    nh.initNode();
```
Comment (emcauliffe) [inline firmware/dmc_stm32/drive_test.ino:55]: please add more info about what is happening here

**Review by RutheniumVI — APPROVED** [review]

**Review by ishanvermani — APPROVED** [review]

**Pull request merged**